locator/network_topology_strategy.cc

```diff
@@ -284,13 +280,6 @@ void network_topology_strategy::validate_options(const gms::feature_service& fs)
     }
 }
 
-std::optional<std::unordered_set<sstring>> network_topology_strategy::recognized_options(const topology& topology) const {
-    // We only allow datacenter names as options
-    auto opts = topology.get_datacenters();
-    opts.merge(recognized_tablet_options());
-    return opts;
```

Comment:
Patch comment says:

> - for NetworkTopology this function returns DC names and tablet options, but tablet
>  options are empty; OTOH this strategy also accepts 'replication_factor'
> tag, which was ommitted,

While "tablet options are empty" is true, the "this strategy also accepts 'replication_factor'" is not